### PR TITLE
[nodejs] update highlighted point in line result

### PR DESCRIPTION
**Pull request by chejennifer**
- pass date in line config
- update the highlighted point in the nodejs line result to be the specified date
- skip returning a line chart result if there is a specified date, but no data point for that date

### CODE REVIEW

File: static/nodejs_server/line_tile.ts [hunk 1]
```diff
@@ -97,15 +97,32 @@ export async function getLineTileResult(
     // If it is a single line chart, add highlight information.
     if (chartData.dataGroup && chartData.dataGroup.length === 1) {
       const dataPoints = _.cloneDeep(chartData.dataGroup[0].value);
-      if (!_.isEmpty(dataPoints)) {
+      const highlightDate = statVarSpec[0].date;
+      let highlightPoint = null;
+      if (!_.isEmpty(dataPoints) && highlightDate) {
+        highlightPoint = dataPoints.find(
+          (point) => point.label === highlightDate
+        );
```
Comment (pradh) [inline static/nodejs_server/line_tile.ts:105]: TODO: consider if we want to allow for monthly stats if year is asked

**Review by pradh — APPROVED** [review]
Thanks!

File: static/nodejs_server/line_tile.ts (hunk 1 above)
Comment (chejennifer) [reply static/nodejs_server/line_tile.ts:107]: added

**Comment by chejennifer** [comment]
thanks for the review!

**Pull request merged**